src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java

```diff
@@ -180,6 +182,36 @@ public boolean canEstimateJobDuration()
     return false;
   }
 
+  /**
+   * 'runway' for laser to get up to speed when rastering (in mm)
+   *
+   */
+  private double rasterPadding = 20;
```

Comment:
Please double-check if this is initialized correctly when loading old settings that do not yet have this property. As far as I remember, XStream sets all missing fields to zero/false/null, ignoring the default value. You may need to implement it similar to here (`Double` instead of `double` datatype, then a missing initialization will be `null` and can be distinguished from `0`). https://github.com/t-oster/LibLaserCut/blob/5215383458060acb7404d01356a2a8af0137f52c/src/main/java/de/thomas_oster/liblasercut/drivers/GenericGcodeDriver.java#L1098-L1104

Reply:
✅

Reply:
As far as I understand, the type should be `Double` not `double`. `Double` can be `null` (like null-pointer), `double` is just a value.

Please check that
- old settings that don't have the value are not loaded as 0 but as 20 (due to the `null` trick).
- in new settings, you can set the value to 0, save it, restart VisiCut, and it still remains at 0.

I'd be surprised if the current code actually works in all these cases.

Reply:
`double` -> `Double` ✅

Reply:
This uppercase D in Double causes compile errors:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project liblasercut: Compilation failure: Compilation failure: 
[ERROR] /home/testy/src/github/kkaempf/LibLaserCut/src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java:[189,34] incompatible types: int cannot be converted to java.lang.Double
[ERROR] /home/testy/src/github/kkaempf/LibLaserCut/src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java:[192,17] getRasterPadding() in de.thomas_oster.liblasercut.drivers.Ruida cannot override getRasterPadding() in de.thomas_oster.liblasercut.LaserCutter
[ERROR]   return type java.lang.Double is not compatible with double
[ERROR] /home/testy/src/github/kkaempf/LibLaserCut/src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java:[191,3] method does not override or implement a method from a supertype
[ERROR] /home/testy/src/github/kkaempf/LibLaserCut/src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java:[193,48] incompatible types: int cannot be converted to java.lang.Double
[ERROR] -> [Help 1]

Reply:
@kkaempf has this been fixed?
It should be enough to change the type the property but leave the functions unchanged.

Reply:
Just merged @jnweiger's PR kkaempf/LibLaserCut#3